src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/YFlowCreateService.java

```diff
@@ -0,0 +1,248 @@
+            @Override
+            public void onCompleted(String flowId) {
+                YFlowCreateContext context = YFlowCreateContext.builder().subFlowId(flowId).build();
+                fsms.forEach((k, fsm) -> {
+                    if (fsm.getPendingSubFlows().contains(flowId)) {
+                        fsmExecutor.fire(fsm, Event.SUB_FLOW_REMOVED, context);
+                    }
+                });
+            }
+
+            public void onFailed(String flowId, String errorReason) {
+                YFlowCreateContext context = YFlowCreateContext.builder()
+                        .subFlowId(flowId)
+                        .error(errorReason)
+                        .build();
+                fsms.forEach((k, fsm) -> {
+                    if (fsm.getPendingSubFlows().contains(flowId)) {
+                        fsmExecutor.fire(fsm, Event.SUB_FLOW_FAILED, context);
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * Handles request for y-flow creation.
+     *
+     * @param key command identifier.
+     * @param request request data.
+     */
+    public void handleRequest(String key, CommandContext commandContext, YFlowRequest request) {
+        log.debug("Handling y-flow create request with key {} and flow ID {}", key, request.getYFlowId());
+
+        if (fsms.containsKey(key)) {
+            log.error("Attempt to create a FSM with key {}, while there's another active FSM with the same key.", key);
+            return;
+        }
+
+        String eventKey = commandContext.getCorrelationId();
+        if (flowEventRepository.existsByTaskId(eventKey)) {
+            log.error("Attempt to reuse key {}, but there's a history record(s) for it.", eventKey);
+            return;
+        }
+
+        if (request.getYFlowId() == null) {
+            request.setYFlowId(UUID.randomUUID().toString());
+        }
+        request.getSubFlows().forEach(subFlow -> {
+            if (subFlow.getFlowId() == null) {
+                subFlow.setFlowId(UUID.randomUUID().toString());
+            }
+        });
+
+        YFlowCreateFsm fsm = fsmFactory.newInstance(commandContext, request.getYFlowId());
+        fsms.put(key, fsm);
+
+        YFlowCreateContext context = YFlowCreateContext.builder()
+                .targetFlow(request)
+                .build();
+        fsm.setTargetFlow(request);
+
+        fsmExecutor.fire(fsm, Event.NEXT, context);
+
+        removeIfFinished(fsm, key);
+    }
+
+    /**
+     * Handles async response from worker.
+     *
+     * @param key command identifier.
+     */
+    public void handleAsyncResponse(String key, SpeakerResponse flowResponse) {
+        log.debug("Received flow command response: {}", flowResponse);
+        YFlowCreateFsm fsm = fsms.get(key);
+        if (fsm == null) {
+            log.debug("Can't find a FSM: received response with key {} for non pending FSM", key);
```

Comment:
In most cases this is error marker, on "debug" level we will never see it. Even when it is not an error I would like to see this message on prod, to be able to measure actual delay between speaker request and speaker response.